alby.go

```diff
@@ -435,6 +436,92 @@ func (svc *AlbyOAuthService) SendPaymentSync(ctx context.Context, senderPubkey,
 	return "", errors.New(errorPayload.Message)
 }
 
+func (svc *AlbyOAuthService) SendKeysend(ctx context.Context, senderPubkey string, amount int64, destination, preimage string, custom_records []TLVRecord) (preImage string, err error) {
+	app := App{}
+	err = svc.db.Preload("User").First(&app, &App{
+		NostrPubkey: senderPubkey,
+	}).Error
+	if err != nil {
+		svc.Logger.WithFields(logrus.Fields{
+			"senderPubkey": senderPubkey,
+			"payeePubkey":  destination,
+		}).Errorf("App not found: %v", err)
+		return "", err
+	}
+	svc.Logger.WithFields(logrus.Fields{
+		"senderPubkey": senderPubkey,
+		"payeePubkey":  destination,
+		"appId":        app.ID,
+		"userId":       app.User.ID,
+	}).Info("Processing payment request")
+	tok, err := svc.FetchUserToken(ctx, app)
+	if err != nil {
+		return "", err
+	}
+	client := svc.oauthConf.Client(ctx, tok)
+
+	customRecordsMap := make(map[string]string)
+	for _, record := range custom_records {
```

Comment:
This conversion would not need to be done in the Alby client if we use a Map<string, string>

Reply:
Edit: sorry I was wrong